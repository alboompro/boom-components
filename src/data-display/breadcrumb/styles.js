import styled from "styled-components";

export const Container = styled.div`
  margin: 40px auto;
  max-width: ${props => (props.small ? "740px;" : "1300px;")};

  

  @media screen and (max-width: 1600px) {
    margin: 40px auto;
  }

  @media screen and (max-width: 767px) {
    margin: 20px;
    padding: 0;
  }
  `;

export const Breadcrumbs = styled.ul`
  list-style: none;
  padding: 0;
  & > li:after {
    padding: 0 8px;
  }
`;

export const Crumb = styled.li`
  display: inline-block;

  &:last-of-type:after {
    content: "";
    padding: 0;
  }

  a {
    text-decoration: none;

    &:hover, &:active {
      text-decoration: underline;
    }
  }
`;