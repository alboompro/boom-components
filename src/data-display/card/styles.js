import styled from "styled-components";

export const CardStyle = styled.div`
  box-sizing: border-box;
  margin-bottom: 10px;
  padding: 0;
  color: #000000d9;
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  position: relative;
  background: #fff;
  border-radius: 2px;
  border: ${StyleProps => (StyleProps.bordered ? "1px solid #f0f0f0" : "none")};

  ${StyleProps =>
    StyleProps.hoverable
      ? `&:hover { cursor: pointer;
        box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;}
        transition: box-shadow 0.3s, border-color 0.3s;
        `
      : ""}
  img {
    width: 100%;
  }
`;

export const HeadStyle = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;
  min-height: ${HeadProps => (HeadProps.size == "small" ? "36px" : "48px")};
  margin-bottom: -1px;
  padding: ${HeadProps => (HeadProps.size == "small" ? "0 12px" : "0 24px")};
  color: #000000d9;
  font-weight: 500;
  font-size: 16px;
  border-bottom: 1px solid #f0f0f0;
  border-radius: 2px 2px 0 0;
`;

export const TitleStyle = styled.div`
  display: inline-block;
  flex: 1;
  padding: ${HeadProps => (HeadProps.size == "small" ? "8px 0px" : "16px 0px")};
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  font-weight: bold;
  font-size: ${HeadProps => (HeadProps.size == "small" ? "14px" : "16px")};
`;

export const ExtraStyle = styled.div`
  float: right;
  margin-left: auto;
  padding: ${HeadProps => (HeadProps.size == "small" ? "8px" : "16px")};
  color: #000000d9;
  font-weight: 400;
  font-size: 14px;
`;

export const ActionStyle = styled.ul`
  display: flex;
  align-items: center;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  background: #fff;
  border-top: 1px solid #f0f0f0;

  & > li {
    float: left;
    margin: 12px 0;
    color: #00000073;
    text-align: -webkit-center;
  }

  & > li:not(:last-child) {
    border-right: 1px solid #f0f0f0;
  }

  &:before {
    display: table;
    content: "";
  }

  &:after {
    display: table;
    clear: both;
    content: "";
  }
`;

export const BodyStyle = styled.div`
  padding: ${BodyProps => (BodyProps.size == "small" ? "12px" : "24px")};

  p {
    margin: 0px;
  }
`;
