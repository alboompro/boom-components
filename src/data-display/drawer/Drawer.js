import React, { Fragment, useEffect, useState, useRef } from "react";
import PropTypes from "prop-types";
import { noop } from "../../helpers";
import Portal from "../../shared/portal";

import {
  Backdrop,
  DrawerStyle,
  DrawerContent,
  DrawerHeader,
  DrawerBody,
  DrawerWrapper,
  DrawerDialog
} from "./styles";

const Drawer = ({
  title,
  footer,
  closeButton,
  onClose,
  children,
  backdrop,
  visible,
  handleClose,
  ...props
}) => {
  const animationdelay = 1000;
  const [debouncedVisible, setDebouncedVisible] = useState(visible);
  const [hide, setHide] = useState(!visible);
  const timeoutRef = useRef(null);

  /* mostrar drawer não tem delay, mas para remover tem um delay (animationdelay),
   * usando o useEffect e visible
   */

  useEffect(() => {
    if (visible) {
      setDebouncedVisible(true);
      setHide(false);
      clearTimeout(timeoutRef.current);
    } else {
      setDebouncedVisible(false);
      timeoutRef.current = setTimeout(() => {
        setHide(true);
      }, animationdelay); // Adicionar prop.delay para controle de animação
    }
  }, [visible]);

  useEffect(() => () => clearTimeout(timeoutRef.current), []);

  const backdropProps = {
    backdropClosable: props.backdropProps,
    backdrop: props.backdrop,
    backdropStyle: props.backdropStyle,
    visible: debouncedVisible
  };

  const StyleProps = {
    zIndex: props.zIndex,
    hoverable: props.hoverable,
    placement: props.placement,
    ...props
  };

  const WrapperProps = {
    width: props.width,
    height: props.height,
    visible: debouncedVisible,
    placement: props.placement,
    ...props
  };

  const renderCloseButton = () => {
    return (
      closeButton && (
        <button
          type="button"
          aria-label="close"
          style={{
            fontSize: "13.5px",
            cursor: "pointer",
            width: "20px",
            background: "none",
            borderStyle: "none"
          }}
          onClick={() => handleClose()}
        >
          X
        </button>
      )
    );
  };

  const renderBackdrop = () => {
    return props.backdropClosable ? (
      <Backdrop {...backdropProps} onClick={() => handleClose()} />
    ) : (
      <Backdrop {...backdropProps} />
    );
  };

  const renderHeader = () => {
    return (
      title &&
      closeButton && (
        <DrawerHeader>
          {title && <h4>{title}</h4>}
          {closeButton && renderCloseButton()}
        </DrawerHeader>
      )
    );
  };

  return (
    !hide && (
      <Portal>
        {
          <DrawerStyle
            {...StyleProps}
            visible={debouncedVisible}
            onClick={handleClose}
            animationdelay={animationdelay}
          >
            {renderBackdrop()}
            <DrawerWrapper {...WrapperProps}>
              <DrawerDialog>
                <DrawerContent>
                  {renderHeader()}
                  <DrawerBody>{children}</DrawerBody>
                </DrawerContent>
              </DrawerDialog>
            </DrawerWrapper>
          </DrawerStyle>
        }
      </Portal>
    )
  );
};

Drawer.propTypes = {
  /** whether to show backdrop or not (area outside of the drawer) */
  backdrop: PropTypes.bool,
  /** whether to click on backdrop should close drawer */
  backdropClosable: PropTypes.bool,
  /** style of backdrop */
  backdropStyle: PropTypes.object,
  /** style of floating layer which is used for adjusting its position */
  bodyStyle: PropTypes.object,
  /** whether a close button is visible on top right corner of drawer */
  closeButton: PropTypes.bool,
  /** whether to unmount child components on close */
  destroyOnClose: PropTypes.bool,
  /** return the mounted node for drawer */
  getContainer: PropTypes.func,
  /** height of drawer while its placement is top or bottom */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** callback when visible state changes */
  onVisibleChange: PropTypes.func,
  /** placement of drawer */
  placement: PropTypes.oneOf(["top", "right", "bottom", "left"]),
  /** title of drawer */
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /** whether drawer is visible or not */
  visible: PropTypes.bool,
  /** CHANGE THIS */
  onVisibleChange: PropTypes.func,
  /** width of drawer while its placement is right or left */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** z-index property of drawer */
  zIndex: PropTypes.number
};

Drawer.defaultProps = {
  backdrop: true,
  backdropClosable: true,
  backdropStyle: {},
  bodyStyle: {},
  closeButton: true,
  destroyOnClose: false,
  getContainer: () => document.body,
  height: 310,
  onVisibleChange: noop,
  placement: "right",
  title: null,
  visible: false,
  handleClose: noop,
  width: 310,
  zIndex: 1000
};

export default Drawer;
