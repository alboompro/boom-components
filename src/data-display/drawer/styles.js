import styled, { keyframes } from "styled-components";

// -> fade animations for the backdrop
const fadeIn = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 0.8;
  }
`;

const fadeOut = keyframes`
  from {
    opacity: 0.8;
  }

  to {
    opacity: 0;
  }
`;

// -> slide animations for the drawer
const leftSlideIn = keyframes`
  from { transform: translateX(-100%); }
  to { transform: translateX(0); }
`;

const rightSlideIn = keyframes`
  from { transform: translateX(100%); }
  to { transform: translateX(0%); }
`;

const topSlideIn = keyframes`
  from { transform: translateY(-100%); }
  to { transform: translateY(0%); }
`;

const bottomSlideIn = keyframes`
  from { transform: translateY(100%); }
  to { transform: translateY(0%); }
`;

const leftSlideOut = keyframes`
  from { transform: translateX(0%); }
  to { transform: translateX(-100%); }
`;

const rightSlideOut = keyframes`
  from { transform: translateX(0%); }
  to { transform: translateX(100%); }
`;

const topSlideOut = keyframes`
  from { transform: translateY(0%); }
  to { transform: translateY(-100%); }
`;

const bottomSlideOut = keyframes`
  from { transform: translateY(0%); }
  to { transform: translateY(100%); }
`;

// Backdrop Style
export const Backdrop = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: auto;
  background-color: #000;
  animation: ${props => (props.visible ? fadeIn : fadeOut)} 0.5s ease-in-out;
  ${props => (props.visible ? "opacity: 0.8;" : "opacity: 0;")}
`;

// Drawer Style
export const DrawerStyle = styled.div`
  position: fixed;
  z-index: ${props => props.zIndex};
  top: 0;
  height: 100%;
  width: 100%;
  color: #000000d9;
`;

export const DrawerWrapper = styled.div`
  // Add size change by prop and swipe location (top and bottom)
  display: block;
  position: absolute;
  ${props => {
    switch (props.placement) {
      case "left":
        return "left: 0";
      case "right":
        return "right: 0";
      case "bottom":
        return "bottom: 0";
      case "top":
        return "top: 0";

      default:
        break;
    }
  }}
  ${props => {
    if (props.placement !== "top" && props.placement !== "bottom")
      return `width: ${props.width}px; height: 100%`;
    return `height: ${props.height}px; width: 100%`;
  }}
  animation: ${props => {
    if (props.visible) {
      switch (props.placement) {
        case "left":
          return leftSlideIn;
        case "right":
          return rightSlideIn;
        case "top":
          return topSlideIn;
        case "bottom":
          return bottomSlideIn;

        default:
          break;
      }
    } else {
      switch (props.placement) {
        case "left":
          return leftSlideOut;
        case "right":
          return rightSlideOut;
        case "top":
          return topSlideOut;
        case "bottom":
          return bottomSlideOut;

        default:
          break;
      }
    }
  }}
    0.7s forwards;
`;

export const DrawerDialog = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
`;

export const DrawerContent = styled.div`
  position: absolute;
  background-color: #fff;
  outline: 0;
  width: 100%;
  height: 100%;

  -webkit-box-shadow: 0 4px 4px rgba(0, 0, 0, 0.12),
    0 0 10px rgba(0, 0, 0, 0.06);
  box-shadow: 0 4px 4px rgba(0, 0, 0, 0.12), 0 0 10px rgba(0, 0, 0, 0.06);
`;

export const DrawerHeader = styled.div`
  margin: 20px 20px 0;
  position: relative;
  display: flex;
  justify-content: space-between;

  h4 {
    font-weight: bolder;
    font-size: 1.5em;
    line-height: 1.25;
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
    margin: 0;
  }
`;

export const DrawerBody = styled.div`
  position: relative;
  overflow: auto;
  margin: 30px 20px;

  ${props => {
    switch (props.placement) {
      case "left":
        return `height: 100vh;`;
      case "right":
        return `height: 100vh;`;
      case "top":
        return `height: calc(${props.height}px - 100px);`; // - 30 - 20 - 20 - 20 - 10
      case "bottom":
        return `height: calc(${props.height}px - 100px);`; // - 30 - 20 - 20 - 20 - 10

      default:
        break;
    }
  }}
`;
