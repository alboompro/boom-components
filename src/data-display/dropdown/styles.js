/* eslint-disable indent */
import styled, { css } from "styled-components";

export const DropdownWrapper = styled.div`
  .dispatcher {
    display: inline-block;
  }
`;

export const DropdownContainer = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
`;

export const DropdownPlacement = styled.div`
  position: absolute;
  z-index: 1080;
`;

export const DropdownContent = styled.div`
  border-radius: 3px;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: rgba(0, 0, 0, 0.25) 0px 0px 6px;
  padding: 20px;
  font-size: 15px;
  font-weight: 400;
  color: #212121;
  text-align: left;
  width: 230px;
  margin-top: 6px;

  &:before {
    position: absolute;
    content: "";
    display: block;
    width: 0;
    height: 0;

    ${props => {
      switch (props.align) {
        case "right":
          return css`
            left: calc(100% - 16px);
          `;
        case "center":
          return css`
            left: 50%;
            transform: translateX(-50%);
          `;
        default:
          return css`
            left: 10px;
          `;
      }
    }}

    z-index: 3;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    ${props => {
      if (props.arrowPos === "top") {
        return css`
          top: 100%
          border-top: 6px solid white;
        `;
      }
      return css`
        top: 0;
        border-bottom: 6px solid white;
      `;
    }}
  }
`;
