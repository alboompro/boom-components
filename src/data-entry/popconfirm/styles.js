import styled from "styled-components";


export const Container = styled.div`
  margin: 40px auto;
  position: relative;
  

  @media screen and (max-width: 1600px) {
    margin: 40px auto;
  }

  @media screen and (max-width: 767px) {
    margin: 20px;
    padding: 0;
  }
  `;

export const Popover = styled.div`
  box-sizing: border-box;
  color: black;
  position: absolute;
  background-color: #fff;
  padding: 1.5rem;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.26);
  width: auto;
  top: ${props => props.position.top};
  right: ${props => props.position.right};
  left: ${props => props.position.left};
  bottom: ${props => props.position.bottom};
  z-index: 5;

  &.popover-show{
    visibility: visible;
  }

  &.popover-hidden{
    visibility: hidden;
  }

  .popover-message{
    display: flex;
    justify-content: center;
    position: relative;
    align-items: center;
  }

  >section{
    display: flex;
    justify-content: flex-end;
  }
`;

export const Button = styled.a`
  width: 50px;
  border-radius: 2px;
  background-color: #6abd7b;
  font-size: 16px;
  font-weight: 500;
  text-align: center;
  color: #ffffff;
  padding: 5px;
  margin-bottom: 15px;
  margin-left: 8px;
  cursor:pointer;

  @media screen and (max-width: 768px) {
    max-width: 280px;
    width: 100%;
  }
`;