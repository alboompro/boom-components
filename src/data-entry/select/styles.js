import styled from "styled-components";

export const SelectContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

  * {
    box-sizing: border-box;
  }
`;

export const Label = styled.span`
  flex: 0;
  margin: 0 0 5px;
  color: #000;
  font-size: 14px;
  line-height: 16px;
`;

export const SelectContent = styled.div`
  flex: 1;
  position: relative;
`;

export const SelectItem = styled.div`
  display: flex;
  align-items: center;
  border: 1px solid #c9c9c9;
  position: relative;
  cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
  height: 30px;
  padding: 0 5px 0 10px;
  font-size: 14px;
  opacity: ${props => (props.disabled ? ".5" : "1")};
  border-radius: ${props =>
    props.roundedBorder ? `${props.roundedBorder}px` : "unset"};

  &:after {
    content: "";
    display: ${props => (props.showArrow ? "block" : "none")};
    width: 0;
    height: 0;
    margin-left: auto;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;

    ${props => `border-${props.open ? "bottom" : "top"}: 5px solid #595959;`}
  }

  ${props => props.selectStyle}
`;

export const OptionContainer = styled.ul`
  position: absolute;
  list-style: none;
  margin: 0;
  padding: 0;
  background-color: #fff;
  border: 1px solid #c9c9c9;
  ${props => props.border && `border-${props.border}: none;`}
  cursor: pointer;
  overflow: auto;
  border-radius: ${({ roundedBorder }) => roundedBorder || "unset"};

  ${props => props.style}
`;

export const OptionScroll = styled.ul`
  overflow-y: auto;
  max-height: ${props => `${props.maxHeight}px`};
`;

export const Option = styled.li`
  display: flex;
  align-items: center;
  height: 30px;
  padding: 0 10px;
  font-size: 14px;
  font-weight: ${props => (props.selected ? "600" : "500")};
  background-color: ${props => (props.hovered ? props.hoverBgColor : "#fff")};
  ${props => props.multiSelected && "color: #b3b3b3"}
`;

export const DropdownBackdrop = styled.div`
  * {
    box-sizing: border-box;
  }

  position: absolute;
  z-index: ${props => props.zIndex};
  top: 0;
  left: 0;
  bottom: 0;
  width: 100%;

  ${props => props.style}
`;

export const Placeholder = styled.span`
  font-size: 12px;
  color: #9c9c9c;
`;

export const ActionsContainer = styled.div``;
