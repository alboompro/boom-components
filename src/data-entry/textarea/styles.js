import styled from "styled-components";
import { DefaultDataEntryCSS } from "../../shared/data-entry-text";

export const Textarea = styled.textarea`
  ${DefaultDataEntryCSS};

  resize: none;
`;
