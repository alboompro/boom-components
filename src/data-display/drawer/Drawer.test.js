import React from "react";
import { mount, shallow } from "enzyme";

import Drawer from "./Drawer.js";
import { Backdrop } from "./styles.js";

describe("Drawer - Snapshot", () => {
  test("Renders the component correctly", () => {
    const wrapper = shallow(
      <Drawer
        title="Drawer title"
        backdrop={false}
        backdropClosable={false}
        backdropStyle={{ backgroundColor: "red" }}
        closeButton={false}
        zIndex={1001}
        animationDelay={777}
        height={450}
        width={600}
        placement="top"
        handleClose={() => console.log("handleClose")}
        visible
      >
        <div>Drawer Content</div>
      </Drawer>
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("Drawer - Behavior", () => {
  test("Values ​​of props are correct", () => {
    const wrapper = mount(
      <Drawer
        title="Drawer title"
        backdrop={false}
        backdropClosable={false}
        backdropStyle={{ backgroundColor: "red" }}
        closeButton={false}
        zIndex={1001}
        animationDelay={777}
        height={450}
        width={600}
        placement="top"
        handleClose={() => console.log("handleClose")}
        visible
      >
        <div>Drawer Content</div>
      </Drawer>
    );

    expect(wrapper.props().visible).toBe(true);
    expect(wrapper.props().title).toBe("Drawer title");
    expect(wrapper.props().closeButton).toBe(false);
    expect(wrapper.props().animationDelay).toBe(777);
    expect(wrapper.props().backdropClosable).toBe(false);
    expect(wrapper.props().backdropStyle).toEqual({ backgroundColor: "red" });
    expect(wrapper.props().zIndex).toBe(1001);
    expect(wrapper.props().placement).toBe("top");
    expect(wrapper.props().width).toBe(600);
    expect(wrapper.props().height).toBe(450);
  });

  test("Click on the backdrop to close the drawer", () => {
    const wrapper = mount(
      <Drawer
        title="Drawer title"
        handleClose={() => console.log("handleClose")}
        visible
      >
        <div>Drawer Content</div>
      </Drawer>
    );

    expect(wrapper.props().visible).toBe(true);

    wrapper.find(Backdrop).simulate("click");
    setTimeout(() => {
      expect(wrapper.find(Drawer).length).toBe(0);
    }, wrapper.props().animationDelay);
  });
});
