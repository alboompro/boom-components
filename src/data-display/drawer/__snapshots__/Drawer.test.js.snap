// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer - Snapshot Renders the component correctly 1`] = `
<Portal>
  <styled.div>
    <styled.div
      animationDelay={777}
      animationdelay={1000}
      backdrop={false}
      backdropClosable={false}
      backdropStyle={
        Object {
          "backgroundColor": "red",
        }
      }
      bodyStyle={Object {}}
      destroyOnClose={false}
      getContainer={[Function]}
      height={450}
      placement="top"
      visible={true}
      width={600}
      zIndex={1001}
    />
    <styled.div
      animationDelay={777}
      backdropClosable={false}
      backdropStyle={
        Object {
          "backgroundColor": "red",
        }
      }
      bodyStyle={Object {}}
      destroyOnClose={false}
      getContainer={[Function]}
      height={450}
      placement="top"
      visible={true}
      width={600}
      zIndex={1001}
    >
      <styled.div>
        <styled.div>
          <styled.div
            animationDelay={777}
            backdropClosable={false}
            backdropStyle={
              Object {
                "backgroundColor": "red",
              }
            }
            bodyStyle={Object {}}
            destroyOnClose={false}
            getContainer={[Function]}
            height={450}
            placement="top"
            style={Object {}}
            width={600}
            zIndex={1001}
          >
            <div>
              Drawer Content
            </div>
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</Portal>
`;
